test/new-e2e/pkg/e2e/suite.go

```diff
@@ -496,6 +497,20 @@ func (bs *BaseSuite[Env]) TearDownSuite() {
 	if bs.params.devMode {
 		return
 	}
+	// If a Kubernetes environment is detected, we dump the cluster state when tearing down the suite
+	//TODO: *environments.Kubernetes string should be replaces by reflect.TypeOf(&environments.Kubernetes{}) once we move types.go to its own package to avoid import cycles
```

Comment:
I believe the cyclic dependency is there because the code should not be present in the suite. Also it does not work with custom provisioners. That's another thing that would be solved with the interface-based environments.